src/server/http/__tests__/xsrf.js

```diff
@@ -29,10 +30,23 @@ describe('xsrf request filter', function () {
 
     await kbnServer.ready();
 
-    const routeMethods = nonDestructiveMethods.filter(method => method !== 'HEAD').concat(destructiveMethods);
     kbnServer.server.route({
       path: '/xsrf/test/route',
-      method: routeMethods,
+      method: nonDestructiveMethods.filter(method => method !== 'HEAD'),
+      handler: function (req, reply) {
+        reply(null, 'ok');
+      }
+    });
+
+    kbnServer.server.route({
+      path: '/xsrf/test/route',
+      method: destructiveMethods,
+      config: {
+        // Disable payload parsing to make HapiJS server accept any content-type header.
```

Comment:
(btw, I started the Kibana dev server with `npm start -- --no-ssl --no-base-path` here, that's why it's against `http` and with no basepath)